src/acl_hal_mmd.cpp

```diff
@@ -965,22 +996,40 @@ void l_get_physical_devices(acl_mmd_dispatch_t *mmd_dispatch,
   mmd_dispatch->aocl_mmd_get_offline_info(AOCL_MMD_VERSION, sizeof(buf), buf,
                                           NULL);
   buf[sizeof(buf) - 1] = 0;
-  mmd_dispatch->mmd_version = atof(buf);
+  mmd_dispatch->mmd_version = l_parse_mmd_version_str(std::string(buf));
+  if (mmd_dispatch->mmd_version < 0) {
+    printf("  Invalid MMD version:     %s\n", buf);
+    printf("Contact the board package support vendor for resolution.\n");
+    fflush(stdout);
+    assert(0);
+  }
   min_MMD_version =
       (!MMDVERSION_LESSTHAN(min_MMD_version, mmd_dispatch->mmd_version))
           ? mmd_dispatch->mmd_version
           : min_MMD_version;
   ACL_HAL_DEBUG_MSG_VERBOSE(1, "HAL : Getting info version: %s\n", buf);
 
-  if (MMDVERSION_LESSTHAN(atof(AOCL_MMD_VERSION_STRING),
+  static double hal_version = 0;
+  if (hal_version == 0) { // Just parse once at start-up
+    hal_version = l_parse_mmd_version_str(AOCL_MMD_VERSION_STRING);
+    if (hal_version < 0) { // This should theoretically never happen
+      printf("  Invalid runtime version: %s\n", AOCL_MMD_VERSION_STRING);
+      fflush(stdout);
+      assert(0);
+    }
+  }
+  if (MMDVERSION_LESSTHAN(hal_version,
                           mmd_dispatch->mmd_version) || // MMD newer than HAL
       MMDVERSION_LESSTHAN(mmd_dispatch->mmd_version,
                           14.0)) // Before this wasn't forward compatible
   {
     printf("  Runtime version: %s\n", AOCL_MMD_VERSION_STRING);
     printf("  MMD version:     %s\n", buf);
+    printf("MMD version is newer than the runtime version! Use the runtime "
+           "with version greater or equal to the MMD version, or contact the "
+           "board support package vendors if mismatch is unexpected.\n");
```

Comment:
@aktole Could you check if the printed message here is okay?

Reply:
Yes, looks good, thanks Sophie